### PR TITLE
Add BLE controller flash driver support

### PR DESCRIPTION
**Pull request by thomasstenersen**
- Fixes kconfig circular dependencies
- Adds support for `FLASH_NRF_DRIVER` choice
- Adds support for BLE controller in samples/drivers/soc_flash_nrf test example

### CODE REVIEW

File: soc/arm/nordic_nrf/Kconfig.defconfig [hunk 1]
```diff
@@ -28,9 +28,6 @@ endif # GPIO
 
 if FLASH
 
-config SOC_FLASH_NRF
```
Comment (rugeGerritsen) [inline soc/arm/nordic_nrf/Kconfig.defconfig:31]: Why did you remove this?

File: samples/drivers/soc_flash_nrf/src/main.c [hunk 2]
```diff
@@ -21,8 +25,31 @@
...
+}
+#endif /* IS_ENABLED(CONFIG_BT_LL_NRFXLIB) */
+
 void main(void)
 {
+#if IS_ENABLED(CONFIG_BT_LL_NRFXLIB)
+    	int err;
+	/* We need to start the BLE Controller in order for the flash driver to
+	 * work. */
+	err = bt_enable(bt_ready);
```
Comment (rugeGerritsen) [inline samples/drivers/soc_flash_nrf/src/main.c]: I don't think it is necessary to enable BLE. The BLE controller is initialized when the hci_driver is initialized. After that point of time you can use the flash APIs

File: drivers/flash/Kconfig [hunk 3]
```diff
@@ -44,9 +44,14 @@ config FLASH_PAGE_LAYOUT
 	help
 	  Enables API for retrieving the layout of flash memory pages.
 
+choice FLASH_NRF_DRIVER
+	prompt "Nordic Semiconductor flash driver"
+  bool
+	default SOC_FLASH_NRF if !BT_LL_NRFXLIB
```
Comment (rugeGerritsen) [inline drivers/flash/Kconfig]: @PavelVPV was able to get in the clock driver without mentioning BT_LL_NRFXLIB. Is it possible to use the same approach?

File: soc/arm/nordic_nrf/Kconfig.defconfig (hunk 1 above)
Comment (thomasstenersen) [reply soc/arm/nordic_nrf/Kconfig.defconfig:31]: Because kconfig will barf out warnings and whatnot about this now that it is a (sub-)choice of drivers. I could remove the whole `if FLASH ... endif` structure from here.

File: samples/drivers/soc_flash_nrf/src/main.c (hunk 2 above)
Comment (thomasstenersen) [reply samples/drivers/soc_flash_nrf/src/main.c]: I assumed so myself, but for some reason this is not the case. Removing this code will cause the example to halt at the first write attempt. But, perhaps there is something else going on that's halting it.

File: drivers/flash/Kconfig (hunk 3 above)
Comment (thomasstenersen) [reply drivers/flash/Kconfig]: I'll give it a go.

- [ ] Try to not mention BT_LL_NRFXLIB

File: samples/drivers/soc_flash_nrf/src/main.c (hunk 2 above)
Comment (rugeGerritsen) [reply samples/drivers/soc_flash_nrf/src/main.c]: I think we need to debug why this happens. It is likely that something is wrong in the stack

Comment (thomasstenersen) [reply samples/drivers/soc_flash_nrf/src/main.c]: :+1: 

- [x] debug why the flash gets stuck when not `bt_enable()` is called

Comment (thomasstenersen) [reply samples/drivers/soc_flash_nrf/src/main.c]: The signalling thread in the HCI driver was started in `hci_driver_open()` which is called by `bt_enable()`, I've moved the starting of the thread to initialization time in the other pull request.

File: drivers/flash/Kconfig (hunk 3 above)
Comment (thomasstenersen) [reply drivers/flash/Kconfig]: I see that both @PavelVPV 's driver and Zephyr's own is included in the compiled binary (I see functions from both). That should not be the case, right? It's an either-or thing?

Comment (PavelVPV) [reply drivers/flash/Kconfig]: @thomasstenersen which functions do you see?

Comment (thomasstenersen) [reply drivers/flash/Kconfig]: (for context) I hadn't merged with the latest changes in `rugeGerritsen:develop/DRGN-10888_adapt_to_new_api`.

Now I see that the changes to kconfig makes the existing entropy and clock control driver dependent on the BT_LL_SW. I think we need to come up with some other way to achieve this.

**Review by rugeGerritsen — APPROVED** [review]
KConfig looks much more clean now

**Review by PavelVPV — APPROVED** [review]
